ctapipe/atmosphere/model.py

```diff
@@ -0,0 +1,250 @@
+
+
+    Attributes
+    ----------
+    h0: u.Quantity["m"]
+        scale height
+    rho0: u.Quantity["g cm-3"]
+        scale density
+    """
+
+    h0: u.Quantity = 8 * u.km
+    rho0: u.Quantity = 0.00125 * u.g / (u.cm**3)
+
+    @u.quantity_input
+    def __call__(self, h: u.m) -> u.Quantity:
+        return self.rho0 * np.exp(-h / self.h0)
+
+    @u.quantity_input
+    def integral(
+        self,
+        h: u.m,
+        output_units=u.g / u.cm**2,
+    ) -> u.Quantity:
+        return self.rho0 * self.h0 * np.exp(-h / self.h0)
+
+
+class TableAtmosphereDensityProfile(AtmosphereDensityProfile):
+    """Tabular profile from a table that has both the density and it's integral
+    pre-computed.  The table is interpolated to return the density and its integral.
+
+    Attributes
+    ----------
+    table: Table
+        Points that define the model
+    """
+
+    def __init__(self, table: Table):
+        """
+        Parameters
+        ----------
+        table: Table
+            Table with columns `height`, `density`, and `column_density`
+        """
+
+        for col in ["height", "density", "column_density"]:
+            if col not in table.colnames:
+                raise ValueError(f"Missing expected column: {col} in table")
+
+        self.table = table
+        self._density_interp = interp1d(
+            table["height"].to("km"), table["density"].to("g cm-3")
+        )
+        self._col_density_interp = interp1d(
+            table["height"].to("km"), table["column_density"].to("g cm-2")
+        )
+
+    @classmethod
+    def from_simtel(
+        cls, simtel_filename: str, profile_number: int = 0
+    ) -> AtmosphereDensityProfile:
+        """Construct a TableAtmosphereDensityProfile from a simtel file
+        containing a set of atmosphere profiles.
+
+        Parameters
+        ----------
+        simtel_filename: str
+            filename of a SimTelArray data file
+        profile_number: int
+            index of profile in the file if there are more than one
+        """
+        return cls(table=read_simtel_profile(simtel_filename)[profile_number])
+
+    @u.quantity_input
+    def __call__(self, h: u.m) -> u.Quantity:
+        return u.Quantity(self._density_interp(h.to_value(u.km)), u.g / u.cm**3)
+
+    @u.quantity_input
+    def integral(self, h: u.m) -> u.Quantity:
+        return u.Quantity(self._col_density_interp(h.to_value(u.km)), u.g / u.cm**2)
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}(meta={self.table.meta}, rows={len(self.table)})"
+        )
+
+
+def read_simtel_profile(simtelfile: str) -> Table:
+    """
+    Read an atmosphere profile from a SimTelArray file as an astropy Table
+
+    Returns
+    -------
+    Table:
+        table with columns `height`, `density`, and `column_density`
+        along with associated metadata
+    """
+    import eventio
+
+    tables = []
+    with eventio.SimTelFile(simtelfile) as simtel:
```

Comment:
Shouldn't this just be in a `SimTelEventSource.atmospheric_profiles` attribute?

Reply:
This is also a recent addition in the mc files, so it has to anyways handle the case that non are available.

Reply:
Indeed, there *should* only be one profile and I only see one in the simtel files.

The read logic in pyeventio just parses the objects and if it encounters an `AtmosphericProfile` object, it appends to the list.

If we are sure, there can only be one, we can change the `SimTelFile` in eventio.